app/controllers/brapi/v1/phenotypes_controller.rb

```diff
@@ -0,0 +1,130 @@
+class Brapi::V1::PhenotypesController < Brapi::BaseController
+
+
+  attr_accessor :request_params, :user
+
+  rescue_from ActionController::ParameterMissing do |exception|
+    render json: { errors: { attribute: exception.param, message: exception.message } }, status: 422
+  end
+
+
+  def search
+    # accepted params: germplasmDbIds, observationVariableDbIds, studyDbIds, locationDbIds, seasonDbIds
+    # sortBy, sortOrder
+    # pageSize, page
+    # TODO Not supported yet: programDbIds, observationLevel
+
+    phenotypes_queries = Brapi::V1::PhenotypesQueries.instance
+    page = get_page
+    page_size = get_page_size 
+        
+    result_object = phenotypes_queries.phenotypes_search_query(
+      params['germplasmDbIds'], params['observationVariableDbIds'], params['studyDbIds'], 
+      params['locationDbIds'], params['seasonDbIds'], 
+      params['sortBy'], params['sortOrder'], page, page_size, false)
+    
+    records = result_object.values
+   
+    if records.nil? || records.size ==0
+      render json: { reason: 'Resource not found' }, status: :not_found  # 404
+    else
+      json_result_array = []
```

Comment:
I think this part could use some restructuring in order to improve readability. Perhaps something like below? This is just a proposition though.

```ruby
observation_units = []
observations = Hash.new { |h, k| h[k] = [] }

result_object.each do |observation|
  observation_unit_id = observation['observationUnitDbId']
  observation_unit = extract_observation_unit(observation)

  if observations[observation_unit_id].empty?
    observation_unit['observations'] = observations[observation_unit_id]
    observation_units << observation_unit
  end 

  # handle design factor transformations...

  observations[observation_unit_id] << observation
end
```